From 2afdbe7c6b3fc08dec7884a074c7f5b8b87a30ad Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Sat, 14 Feb 2026 16:05:16 +0100
Subject: [PATCH] CalendarEvents: Avoid Posix.NLTime if not available

_NL_TIME_FIRST_WEEKDAY and similar are glibc-ism's and don't exist on other libc
implementations like musl libc.

So if it's not available, set first weekday to 0 instead.

Signed-off-by: Achill Gilgenast <achill@achill.org>
---
 core/Services/CalendarEvents/CalendarEvents.vala |  4 ++++
 meson.build                                      | 13 +++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/core/Services/CalendarEvents/CalendarEvents.vala b/core/Services/CalendarEvents/CalendarEvents.vala
index 1bbcae8fae1b..20d8cf929d5f 100644
--- a/core/Services/CalendarEvents/CalendarEvents.vala
+++ b/core/Services/CalendarEvents/CalendarEvents.vala
@@ -65,7 +65,11 @@ public class Services.CalendarEvents : Object {
         source_components = new HashTable<E.Source, Gee.TreeMultiMap<string, ECal.Component>> (CalendarEventsUtil.source_hash_func, CalendarEventsUtil.source_equal_func);
         source_view = new HashTable<string, ECal.ClientView> (str_hash, str_equal);
 
+#if HAVE__NL_TIME_FIRST_WEEKDAY
         int week_start = Posix.NLTime.FIRST_WEEKDAY.to_string ().data[0];
+#else
+        int week_start = 0;
+#endif
         if (week_start >= 1 && week_start <= 7) {
             week_starts_on = (GLib.DateWeekday) (week_start - 1);
         }
diff --git a/meson.build b/meson.build
index fe833d9075d2..6c5bad81e394 100644
--- a/meson.build
+++ b/meson.build
@@ -4,6 +4,8 @@ project(
   version: '4.18.0'
 )
 
+cc = meson.get_compiler('c')
+
 gnome = import('gnome')
 i18n = import('i18n')
 pkgconfig = import('pkgconfig')
@@ -102,6 +104,17 @@ grep = find_program('grep')
 total_strings_res = run_command(grep, '-c', 'msgid', meson.project_source_root() + '/po/io.github.alainm23.planify.pot', check : true)
 total_strings = total_strings_res.stdout().strip().to_int() - 1
 
+# _NL_TIME_FIRST_WEEKDAY is a glibc-ism
+nl_time_first_weekday_src = '''
+  #include <langinfo.h>
+  int main() {
+    nl_langinfo(_NL_TIME_FIRST_WEEKDAY);
+  };
+'''
+if cc.compiles(nl_time_first_weekday_src)
+    add_project_arguments('-D', 'HAVE__NL_TIME_FIRST_WEEKDAY', language: 'vala')
+endif
+
 conf_data = configuration_data()
 conf_data.set('TOTAL_STRINGS', total_strings)
 conf_data.set_quoted('APPLICATION_ID', application_id)
-- 
2.53.0

